core/src/androidMain/kotlin/BluetoothDeviceAndroidPeripheral.kt

```diff
@@ -69,22 +68,14 @@ internal class BluetoothDeviceAndroidPeripheral(
 
     override val identifier: String = bluetoothDevice.address
 
-    private val job = SupervisorJob(parentCoroutineContext[Job]).apply {
-        invokeOnCompletion {
-            close()
-            threading.close()
-        }
-    }
-    private val scope = CoroutineScope(parentCoroutineContext + job)
+    private val scope = CoroutineScope(
+        parentCoroutineContext +
+            SupervisorJob(parentCoroutineContext.job).apply {
+                invokeOnCompletion(::dispose)
+            },
+    )
 
-    init {
-        bluetoothState
-            .filter { state -> state == STATE_OFF }
-            .onEach { disconnect() }
-            .launchIn(scope)
-    }
```

Comment:
This was moved within `establishConnection`, so it will now match the lifecycle of a connection rather than living for the entire life of the peripheral.

This is a step in the process of resolving #378.